sdk/trace/span_processor_annotator_example_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/export/trace/tracetest"
+)
+
+/*
+Sometimes information about a runtime environment can change dynamically or be
+delayed from startup. Instead of continuously recreating and distributing a
+TracerProvider with an immutable Resource or delaying the startup of your
+application on a slow loading piece of information, annotate the created spans
+dynamically using a SpanProcessor.
+*/
+
+// AttrsFunc is called when annotations for a Span need to be determined.
+type AttrsFunc func(context.Context) []attribute.KeyValue
+
+// Annotator is a SpanProcessor that adds attributes to all started spans.
+type Annotator struct {
+	// Next is the next SpanProcessor in the chain.
+	Next SpanProcessor
```

Comment:
It's not necessary for each processor to call on a next one in order to form a pipeline. The `TracerProvider` already sets up such a sequential pipeline.

Reply:
We will need this for the end/export part of the processing. I've renamed to make it clearer.